nucypher/blockchain/eth/signers.py

```diff
@@ -194,3 +220,93 @@ def unlock_account(self, account: str, password: str, duration: int = None) -> b
     @validate_checksum_address
     def lock_account(self, account: str):
         return True
+
+
+class KeyStoreSigner(Signer):
+
+    IS_SIGNER_LOCAL = True
+
+    def __init__(self, path: str):
+        super().__init__()
+
+        with open(path) as file_obj:
+            key = json.load(file_obj)
+
+        address = key['address']
+        if not is_address(address):
+            raise ValueError(f"'{path}' does not contain a valid address")
+
+        self.__key = key
+        self.__account = to_checksum_address(address)
+        self.__signer = None
+
+    @classmethod
+    def from_signer_uri(cls, uri: str) -> 'Signer':
+        decoded_uri = urlparse(uri)
+
+        if decoded_uri.scheme not in ('key', 'keystore') or decoded_uri.netloc:
+            raise cls.InvalidSignerURI(uri)
+
+        return cls(path=decoded_uri.path)
+
+    @validate_checksum_address
+    def is_device(self, account: str) -> bool:
+        return False
+
+    def accounts(self) -> List[str]:
+        return [self.__account]
+
+    @validate_checksum_address
+    def unlock_account(self, account: str, password: str, duration: int = None) -> bool:
+        if self.__account != account:
+            return False
+
+        if not self.__signer:
+            try:
+                self.__signer = Account.from_key(
+                    Account.decrypt(self.__key, password)
+                )
+            except ValueError:
+                return False
+
+            if self.__account != self.__signer.address:
+                raise ValueError(f"unexpected address '{self.__signer.address}' recovered from private key; expected '{self.__account}'")
```

Comment:
Since `unlock_account` failed, should we set `self.__signer` to `None` (i.e. what it was prior to the function call) before raising the error?

Reply:
That seems reasonable.